### ISSUE TITLE
API not answering when using 2x proxy and SSL in some rare case

### ISSUE THREAD

**Issue by metasnw** [metasfresh/metasfresh-webui-api#471]
### Is this a bug or feature request?
bug

### What is the current behavior?
API stops responding leading to empty windows

restarting the API fixes the problem

log:
```
2017-06-23 15:53:53.053 ERROR 15 --- [http-nio-8080-exec-5] o.a.c.c.C.[Tomcat].[localhost]           : Exception Processing ErrorPage[errorCode=0, location=/error]

org.apache.catalina.connector.ClientAbortException: java.io.IOException: Broken pipe
        at org.apache.catalina.connector.OutputBuffer.doFlush(OutputBuffer.java:321)
        at org.apache.catalina.connector.OutputBuffer.flush(OutputBuffer.java:284)
        at org.apache.catalina.connector.CoyoteOutputStream.flush(CoyoteOutputStream.java:118)
        at org.springframework.session.web.http.OnCommittedResponseWrapper$SaveContextServletOutputStream.flush(OnCommittedResponseWrapper.java:458)
        at org.springframework.security.web.util.OnCommittedResponseWrapper$SaveContextServletOutputStream.flush(OnCommittedResponseWrapper.java:514)
        at com.fasterxml.jackson.core.json.UTF8JsonGenerator.flush(UTF8JsonGenerator.java:1054)
        at com.fasterxml.jackson.databind.ObjectWriter.writeValue(ObjectWriter.java:953)
        at org.springframework.http.converter.json.AbstractJackson2HttpMessageConverter.writeInternal(AbstractJackson2HttpMessageConverter.java:279)
        at org.springframework.http.converter.AbstractGenericHttpMessageConverter.write(AbstractGenericHttpMessageConverter.java:106)
        at org.springframework.web.servlet.mvc.method.annotation.AbstractMessageConverterMethodProcessor.writeWithMessageConverters(AbstractMessageConverterMethodProcessor.java:231)
        at org.springframework.web.servlet.mvc.method.annotation.HttpEntityMethodProcessor.handleReturnValue(HttpEntityMethodProcessor.java:203)
        at org.springframework.web.method.support.HandlerMethodReturnValueHandlerComposite.handleReturnValue(HandlerMethodReturnValueHandlerComposite.java:81)
        at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:113)
        at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:827)
        at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:738)
        at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:85)
        at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:963)
        at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:897)
        at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:970)
        at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:872)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:661)
        at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:846)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:742)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:101)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at de.metas.ui.web.config.WebConfig$1.doFilter(WebConfig.java:82)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at de.metas.ui.web.config.ServletLoggingFilter.doFilter(ServletLoggingFilter.java:89)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)

```

#### Which are the steps to reproduce?
opening /window/244/540005 on a particular system.
was not able to reproduce somewhere else yet

### What is the expected or desired behavior?
no crash

**Comment by metasnw** [comment]
system keeps crashing. I updated the trace log

**Comment by teosarca** [comment]
hi @metasnw ,

next time when it happens, pls upload the whole log (via pastebin or something).

**Comment by metasnw** [comment]
@teosarca 
case: 
1. open request window
1. have a request
1. zoom into request type 
1. switch to grid view
1. NOK: API is gone
1. full log: https://drive.google.com/file/d/0BynPmm6FxbR-OExIYW9SRHNGVkk/view?usp=sharing

**Comment by teosarca** [comment]
@metasnw  On a quick look found followings:

It seems that suddenly tomcat starts throwing "IOException: Broken pipe". Usually that happens when the client (browser, proxy) it's closing the connection where it's request is still processed.
Would it be possible/plausible the issue to be outside of our "webui jar"? 

----
Maybe related, but maybe,
I saw that the webui-api server tries to start an ActiveMQ embedded broker but it fails because the port is already taken (which is OK!, read forward).
That's fine because that port shall be used by the embedded broker which sits in metasfresh server.
The weird thing is that webui-api it's trying to start an embedded broker. It shall not.
Can u pls check why (config, sysconfig ...)?

```
java.io.IOException: Transport Connector could not be registered in JMX: java.io.IOException: Failed to bind to server socket: tcp://app:61616 due to: java.net.BindException: Cannot assign requested address (Bind failed)
```

**Comment by metas-ts** [comment]
might be related to https://github.com/metasfresh/metasfresh/issues/1578

but the strange thing is that we have this https://github.com/metasfresh/metasfresh-webui-api/commit/5444bc47c37a5971a23c5c65a7c31887d3fe975b

in the webui-api's applicaiton.properties file

**Comment by teosarca** [comment]
also I observed that the spring context is refreshed quite frequently, e.g.
```
2017-06-26 21:30:46.244  INFO 16 --- [main] ationConfigEmbeddedWebApplicationContext : Refreshing org.springframework.boot.context.embedded.AnnotationConfigEmbeddedWebApplicationContext@e350b40: startup date [Mon Jun 26 21:30:46 CEST 2017]; root of context hierarchy
```

**Comment by metas-ts** [comment]
> But what it's triggering that context refresh...

brainfart of the day: 
maybe the  access timestamp of `application.properties` is changed, because...docker...cloud...stuff?

**Comment by metasnw** [comment]
I could reproduce it on another machine live49 with version 5.15 and an empty database

**Comment by metasnw** [comment]
after excluding different possible causes teo and I came to this conclusion:
as soon as the API goes through a second reverse proxy mixing http and https the problem occurs.
if you leave out ssl and proxy and connect directly it does not happen.

the case also is not related to zoom into but to open another tab accessing that window.
we could reproduce it without zoom by opening a nother tab with the url of the window.

also we found the API is not crashing but the browser can not access it via the proxy servers.

proposed solution:
1. configure API to ssl
1. connect frontend proxy directly to API and spare the second one
